src/codec.rs

```diff
@@ -132,28 +130,27 @@ impl<T> UviBytes<T> {
 
     fn deserialise(&mut self, src: &mut BytesMut) -> Result<Option<BytesMut>, io::Error> {
         loop {
-            match self.len.take() {
+            return match self.len.take() {
```

Comment:
I would prefer not to make the proposed changes to `deserialise`. While functionally equivalent, stylistically they do not read as an improvement to me.

Reply:
ok, done.